lib/query-builder.js

```diff
@@ -9,425 +9,442 @@ var operatorMap = {
-  }
+  function parseExpression (entp, sTypes, current_name, current_value, params, values) {
+    var result = {
+      processed: true
+    }
 
-  if (current_value === null) {
-    // we can't use the equality on null because NULL != NULL
-    params.push(sTypes.escape + RelationalStore.escapeStr(RelationalStore.camelToSnakeCase(current_name)) + sTypes.escape + ' IS NULL')
-    return result
-  }
+    if (current_name.indexOf('$') !== -1) {
+      return result
+    }
 
-  if (current_value instanceof RegExp) {
-    var op = (current_value.ignoreCase) ? '~*' : '~'
-    values.push(current_value.source)
-    params.push(sTypes.escape + RelationalStore.escapeStr(RelationalStore.camelToSnakeCase(current_name)) + sTypes.escape + op + RelationalStore.preparedStatements(sTypes.name, values.length))
-    return result
-  }
+    if (current_value === null) {
+      // we can't use the equality on null because NULL != NULL
+      params.push(sTypes.escape + RelationalStore.escapeStr(toColumn(current_name)) + sTypes.escape + ' IS NULL')
+      return result
+    }
 
-  if (_.isObject(current_value)) {
-    result.processed = false
-    return result
-  }
-  else {
-    values.push(current_value)
-    params.push(sTypes.escape + RelationalStore.escapeStr(RelationalStore.camelToSnakeCase(current_name)) + sTypes.escape + '=' + RelationalStore.preparedStatements(sTypes.name, values.length))
-    return result
-  }
+    if (current_value instanceof RegExp) {
+      var op = (current_value.ignoreCase) ? '~*' : '~'
+      values.push(current_value.source)
+      params.push(sTypes.escape + RelationalStore.escapeStr(toColumn(current_name)) + sTypes.escape + op + RelationalStore.preparedStatements(sTypes.name, values.length))
+      return result
+    }
 
-  return result
-}
+    if (_.isObject(current_value)) {
+      result.processed = false
+      return result
+    }
+    else {
+      values.push(current_value)
+      params.push(sTypes.escape + RelationalStore.escapeStr(toColumn(current_name)) + sTypes.escape + '=' + RelationalStore.preparedStatements(sTypes.name, values.length))
+      return result
+    }
 
-function parseOperation (entp, sTypes, current_name, current_value, params, values, operator, parseExpression) {
-  if (!_.isArray(current_value)) {
-    return {err: operator + ' operator requires an array value'}
+    return result
   }
 
-  var results = []
-  for (var i in current_value) {
-    var w = whereargs(entp, current_value[i])
-    var subParams = []
-    for (var j in w.params) {
-      if (operatorMap[w.params[j]]) {
-        parseOperation(entp, sTypes, w.params[j], w.values[j], subParams, values, operatorMap[w.params[j]], parseExpression)
+  function parseOperation (entp, sTypes, current_name, current_value, params, values, operator, parseExpression) {
+    if (!_.isArray(current_value)) {
+      return {err: operator + ' operator requires an array value'}
+    }
+
+    var results = []
+    for (var i in current_value) {
```

Comment:
`forEach` would add readability here (will replace that ugly `current_value[i]`)